coderd/agentapi/boundary_logs.go

```diff
@@ -2,14 +2,51 @@ package agentapi
 
 import (
 	"context"
+	"time"
 
-	"golang.org/x/xerrors"
+	"github.com/google/uuid"
+
+	"cdr.dev/slog"
 
 	agentproto "github.com/coder/coder/v2/agent/proto"
 )
 
-type BoundaryLogsAPI struct{}
+type BoundaryLogsAPI struct {
+	Log         slog.Logger
+	WorkspaceID uuid.UUID
+}
+
+func (a *BoundaryLogsAPI) ReportBoundaryLogs(ctx context.Context, req *agentproto.ReportBoundaryLogsRequest) (*agentproto.ReportBoundaryLogsResponse, error) {
+	for _, l := range req.Logs {
+		var logTime time.Time
+		if l.Time != nil {
+			logTime = l.Time.AsTime()
+		}
+
+		switch r := l.Resource.(type) {
+		case *agentproto.BoundaryLog_HttpRequest_:
+			if r.HttpRequest == nil {
+				continue
+			}
+			if l.Allowed {
+				a.Log.Info(ctx, "boundary request allowed",
+					slog.F("workspace_id", a.WorkspaceID.String()),
+					slog.F("http_method", r.HttpRequest.Method),
+					slog.F("http_url", r.HttpRequest.Url),
+					slog.F("event_time", logTime.Format(time.RFC3339Nano)),
+				)
+			} else {
+				a.Log.Warn(ctx, "boundary request denied",
```

Comment:
structure the allow/deny better and build a list of these fields once to dedupe the if/else